test/infrastructure/docker/internal/controllers/dockermachine_controller.go

```diff
@@ -312,7 +340,10 @@ func (r *DockerMachineReconciler) reconcileNormal(ctx context.Context, cluster *
 		// but bootstrapped is never set on the object. We only try to bootstrap if the machine
 		// is not already bootstrapped.
 		if err := externalMachine.CheckForBootstrapSuccess(timeoutCtx, false); err != nil {
-			bootstrapData, format, err := r.getBootstrapData(timeoutCtx, machine)
+			if dataSecretName == nil {
```

Comment:
I can't really think of one, but the original implementation of `getBoostrapData()` [here](https://github.com/kubernetes-sigs/cluster-api/blob/022ccf1dd0c240652b115fdb9769cda70ad1be49/test/infrastructure/docker/exp/internal/docker/nodepool.go#L367-L369) is set to return an error if the dataSecretName is nil so I assumed it could be the case if the field was deleted or something. I refactored `getBootstrapData()` to take the dataSecretName as a string parameter instead of the Machine/MachinePool so we could reuse that function.

Reply:
then write this in a comment.
// This condition should not happen at this point, but we are implementing the check for extra safery / something else...